subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1569,6 +1578,14 @@ static int init_reset(void)
 		lll_adv_data_init(&ll_adv[handle].lll.scan_rsp);
 	}
 
+	/* Make sure that set #0 is initialized with empty legacy PDUs. This is
+	 * especially important if legacy HCI interface is used for advertising
+	 * because it allows to enable advertising without any configuration,
+	 * thus we need to have PDUs already initialized.
+	 */
+	init_pdu(lll_adv_data_peek(&ll_adv[0].lll), PDU_ADV_TYPE_ADV_IND);
+	init_pdu(lll_adv_scan_rsp_peek(&ll_adv[0].lll), PDU_ADV_TYPE_SCAN_RSP);
+
```

Comment:
This change is redundant (added CPU use on HCI reset) in Zephyr BT Host usage, as advertising parameters are always set after HCI reset before enabling advertising. Most third party hosts too do that same. (Also, currently there is not default advertising parameters initialized that would allow proper advertising enable without parameters being set by host).

Reply:
I will also handle default parameters on reset and put it inside #ifdef so it's not used with Zephyr host. Regardless of what hosts do, some qualification test cases do not set parameters explicitly and will fail randomly if we don't initialize parameters properly.